addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/InformationDisclosureSuspiciousCommentsScanRule.java

```diff
@@ -57,12 +52,32 @@ public class InformationDisclosureSuspiciousCommentsScanRule extends PluginPassi
 
     private static final int MAX_ELEMENT_CHRS_TO_REPORT = 128;
 
-    public static final String suspiciousCommentsListDir = "xml";
-    public static final String suspiciousCommentsListFile = "suspicious-comments.txt";
-    private static final Logger LOGGER =
-            LogManager.getLogger(InformationDisclosureSuspiciousCommentsScanRule.class);
-
-    private static List<Pattern> patterns = null;
+    public static final String CUSTOM_PAYLOAD_CATEGORY = "Suspicious-Comments";
+    public static final List<String> DEFAULT_PAYLOADS =
+            List.of(
+                    "TODO",
+                    "FIXME",
+                    "BUG",
+                    "BUGS",
+                    "XXX",
+                    "QUERY",
+                    "DB",
+                    "ADMIN",
+                    "ADMINISTRATOR",
+                    "USER",
+                    "USERNAME",
+                    "SELECT",
+                    "WHERE",
+                    "FROM",
+                    "LATER",
+                    "DEBUG");
+
+    private static final Supplier<Iterable<String>> DEFAULT_PAYLOAD_PROVIDER =
+            () -> DEFAULT_PAYLOADS;
+
+    private static Supplier<Iterable<String>> payloadProvider = DEFAULT_PAYLOAD_PROVIDER;
+
+    private List<Pattern> patterns = null;
```

Comment:
This now means that the patterns will compiled for each message scanned. Was that on purpose? Is that to ensure the latest payloads are always used?

Reply:
> Was that on purpose? Is that to ensure the latest payloads are always used?

Yes

Did not want to use premature optimization by using some sort of pattern-cache since the patterns will be easy and the number of patterns will probably not go into millions.
But I respect that there was a static pre-compiled pattern list before the change so if we want to retain this "feature", I am open for further code changes